biz.aQute.repository/src/aQute/bnd/repository/maven/provider/MavenBndRepository.java

```diff
@@ -194,7 +194,7 @@ public PutResult put(InputStream stream, PutOptions options) throws Exception {
 				}
 
 				logger.debug("Put release {}", pom.getRevision());
-				try (Release releaser = storage.release(pom.getRevision(), options.context.getProperties())) {
+				try (Release releaser = storage.release(pom.getRevision(), options.context.getFlattenedProperties())) {
```

Comment:
I think this change should be in a separate commit from the other changes to repo finding since they are totally unrelated. This enable proper cherry-picking if we want.